CRM/Civicase/Form/Report/BaseExtendedReport.php

```diff
@@ -746,6 +975,10 @@ protected function getCustomFieldOptions(array $spec) {
       return $this->getDateColumnOptions($spec);
     }
 
+    if (CRM_Utils_Array::value('data_type', $spec) === 'ContactReference') {
+      return $this->getContactColumnOptions($spec);
```

Comment:
Thinking about this now and wondering if we actually need the `getContactColumnOptions` function. This might end up pulling a lot of contacts from the db and time to run the query and process might have performance issues for sites with large data. What we need actually is the used options. We should probably just set this to an empty array and in the `removeUnusedOptions` function, we can rename this to `getusedOptions` and just return used options without needing to compare with original options because in reality, we are only concerned with used options. In the function we can add a special function for ContactReference fields where we build the option with id as key and display name as value for that case only.

Contacts could get to 1000 or more in some sites and we don't want to fetch all just to build the options when we are only concerned about used ones.

Reply:
yeah that is a valid point but we will still need this function to get the display name as from the used options we only get the ids so I have changed the code to handle the contact ref type field differently so it only loads the used contacts now. kindly check if the changes seems ok